### ISSUE TITLE
Error: tus: unexpected response while uploading chunk, originated from request (response code: 408, response text: stream timeout)

### ISSUE THREAD

**Issue by implicit-invocation** [tus/tus-js-client#176]
**Describe the bug**
I tried using tus-js-client for uploading videos to CloudFlare Stream. Video files are about 1~5GB. Most of the times (49/50), I got this this error

```
Error: tus: unexpected response while uploading chunk, originated from request (response code: 408, response text: stream timeout)
```

**To Reproduce**
1. Use the sample source code from CloudFlare Stream documentation, similar to the one on Readme page.
2. Run from nodejs.
3. See error.

**Expected behavior**
`onSuccess` fired

**Setup details**
Please provide following details, if applicable to your situation:
- Runtime environment: Node.js v12.13.0
- Used tus-js-client version: 1.8.0
- Used tus server software: CloudFlare

**Comment by Acconut** [comment]
Good point, tus-js-client currently does not retry when the server returns a 408 status code (see https://github.com/tus/tus-js-client/blob/master/lib/upload.js#L189-L199), which does not seem correct in this case. Would mind opening a PR to fix this and allow tus-js-client to also retry 408s?

**Comment by rnakashiro** [comment]
Hi, we have some error to upload to CloudFlare Stream, with python tus client.

Current python implementation does not check Status Code. It checks only response header `upload-offset`.
We are happy if the retry checks are fixed.

And once we could upload to CFStream with retry option below 😄 

``` python
        uploader = tus_client.uploader(
            file_path=...,
            chunk_size=...,
            retries=TUS_RETRY_COUNT, # int > 0
            retry_delay=...,
        )
```

**Comment by Acconut** [comment]
@rnakashiro Thanks for reporting, but please open an issue about this in the tus-py-client repository (https://github.com/tus/tus-py-client).

**Comment by rnakashiro** [comment]
@implicit-invocation 
Hi, I have same issue for CFStream, and posted my measurement to cloudflare community. FYI 😄 

https://community.cloudflare.com/t/cloudflare-stream-uploading-with-tus-stops-with-408/130639/6

**Comment by Acconut** [comment]
@rnakashiro Please open an issue in tus-py-client about your problem, so it can be resolved there. This repository is for tus-js-client which you don't seem the be using.

**Comment by rnakashiro** [comment]
@Acconut 
Thanks, as I wrote, I think that should be server side problem.

So, now I don't wanna to be fixed in client library side.
Please make issue yourself if you improve it.

**Comment by Acconut** [comment]
> I think that should be server side problem. So, now I don't wanna to be fixed in client library side.

Sure enough! I am just not sure if CloudFlare is going to patch their servers. That's the impression I get from reading https://community.cloudflare.com/t/cloudflare-stream-uploading-with-tus-stops-with-408/130639

**Comment by rnakashiro** [comment]
@Acconut 
Thanks. FYI below.

The ticket on community don't have reply now, but I made support ticket to CloudFlare about the issue and condition to reproduce.

Then, I could hear that they are investigating the issue of last chunk upload.
I think that if the improvement is released, all client lib will work on JS and python and so on.

**Comment by ifedapoolarewaju** [comment]
> Current python implementation does not check Status Code. It checks only response header upload-offset

@rnakashiro the tuspy client always checks for status code after each request [here](https://github.com/tus/tus-py-client/blob/master/tusclient/uploader.py#L243-L251). Or is it something else you were referring to? 🤔

**Comment by rnakashiro** [comment]
@ifedapoolarewaju 
I'm sorry, I wanted to say about error condition for 40X, such like 403 => destroy, 408 => some retry. 😄

**Comment by frhnfrq** [comment]
Is there any update on this? On the CF community forum post (linked above), I see that the issue is caused by the last chunk taking long time to upload. Someone from CT team has replied that they have made optimizations and we should try to upload the last chunk again when 400 errors are returned

**Comment by kvz** [comment]
Can you ask the CT team to weigh in and perhaps contribute their fix back to this open source project?

**Comment by frhnfrq** [comment]
Created a new ticket on CF support, let's see what they say.

**Comment by daghendrik** [comment]
> 
> 
> Created a new ticket on CF support, let's see what they say.

Did you hear anything back from them, @frhnfrq ?
